lib/semian/net_http.rb

```diff
@@ -0,0 +1,85 @@
+require 'semian'
+require 'semian/adapter'
+require 'net/https'
+require 'open-uri'
+
+module Net
+  ::Net::ProtocolError.include(::Semian::AdapterError)
```

Comment:
nitpick: I've never seen it done like this, seems a little bit unconventional. I would explicitly open the class... but no big deal.